src/Columns/ColumnString.h

```diff
@@ -136,8 +141,12 @@ class ColumnString final : public COWHelper<IColumnHelper<ColumnString>, ColumnS
         const String & s = x.safeGet<String>();
         const size_t old_size = chars.size();
         const size_t size_to_append = s.size() + 1;
-        const size_t new_size = old_size + size_to_append;
 
+        /// First check if new_size won't overflow before resizing.
+        if (size_to_append > std::numeric_limits<size_t>::max() - old_size)
+            throw Exception(ErrorCodes::LOGICAL_ERROR, "ColumnString Field size too large for resizing, it will overflow.");
```

Comment:
maybe `LOGICAL_ERROR` is unnecessary here - maybe `TOO_LARGE_ARRAY_SIZE` or `TOO_LARGE_STRING_SIZE` is better